crowbar_framework/app/models/node_object.rb

```diff
@@ -563,7 +563,12 @@ def asset_tag
     if virtual?
       "vm-#{mac.gsub(':',"-")}"
     else
-      @node[:dmi]["chassis"]["serial_number"] rescue nil
+      asset = @node[:dmi]["chassis"]["serial_number"] rescue nil
+      if @node[:dmi]["chassis"].has_key?("asset_tag") && !@node[:dmi]["chassis"]["asset_tag"].empty?
+        "#{asset} (#{@node[:dmi]["chassis"]["asset_tag"]})"
+      else
+        asset
+      end
```

Comment:
@tboerger what bothers me more is `a["foo"]["bar"]["baz"] rescue nil` sprinkled around *Objects. 

Wouldn't hashie's DeepFind help us here? We could have a static, declared map of "method_name" => [lookup, default value]", then loop over them and use define method? And/or use Mash + bang! methods for the setters?

Or at least we might want to split the accessors to individual 'theme' related modules (i.e RaidAttributes, NetworkAttributes).